gizmo.py

```diff
@@ -0,0 +1,82 @@
+def hello(name, country=None):
+    
+    assert isinstance(name, str),"You forgot your name! unless you are Elon's baby, then sorry write your nickname"
+    if (country!=None):
+        assert isinstance(country,str),"hmmm... that is not a country, please write a valid country"
+    else:
+        country="Finland"
+    print("Hello %s, how are things in %s?" % (name, country))
+
+def spell(word):
+    assert isinstance(word, str),"please enter a valid word, you know... with letters"
+    for i, letter in enumerate(word):
+        if i == len(word) - 1:
+            print(letter)
+        else:
+            print(letter+".", end="")
+
+def relative_path(subject_identifiers):
+    names = []
+    for identifier in subject_identifiers:
+        name = f"./subjects/mock_recording_{identifier}.rec"
+        names.append(name)
+    return names
+
+class Gizmo(object):
+    def __init__(self, name):  
+        self.name = name  
+    
+    def speak(self):  
+        print(self.name)  
+
+def multiplication_table(zero_out_multiples=None):
+    """ Returns a multiplication table form 1 to 12
+    
+    Parameters
+    ----------
+    zero_out_multiples : int, optional
+                        multiples of this parameter will be deleted
+    
+    Returns
+    -------
+    array : array
+            array type with the multiplication table from 1 to 12. If specified
+            the multiples of zero_out_multiples are set to 0.
+    """
+    if (zero_out_multiples!=None):
+        assert isinstance(zero_out_multiples,int),"please enter an integer"
+    else:
+        zero_out_multiples=0
```

Comment:
This calls for a comment why None is translated to 0. I think it's because you now also elegantly deal with divide by zero errors, but this only becomes apparent later on.

Reply:
hehe yeah... I was avoiding comments, but it is true.